### PR TITLE
Add docker dev net setup

### PR DESCRIPTION
**Pull request by lmoe**
This PR introduces a simple/lite devnet setup based on Docker Compose that enables users to spawn a Wasp+GoShimmer combination with one simple `docker-compose up`. 

It is expected to not require any further configuration to make the first steps as easy as possible.

One addition to the setup is the possibility to disable the admin whitelist as it didn't support Docker internal networking so well if the admin port is published to the outside.

### CODE REVIEW

File: packages/webapi/admapi/endpoints.go [hunk 1]
```diff
@@ -36,7 +37,11 @@ func AddEndpoints(
 ) {
 	initLogger()
 
-	adm.EchoGroup().Use(protected(adminWhitelist))
+	echoGroup := adm.EchoGroup()
+
+	if !parameters.GetBool(parameters.WebAPIAdminWhitelistDisabled) {
```
Comment (jorgemmsilva) [inline packages/webapi/admapi/endpoints.go]: this is hard to read, usually is more understandable to make bool the "enabled" and do `!somethingEnalbed`

Comment (lmoe) [reply packages/webapi/admapi/endpoints.go]: Yeah I would generally agree. I just wanted to keep it in the same way as the other config params like:

* LoggerDisableEvents
* LoggerDisableStacktrace
* LoggerDisableCaller

How about something like putting it into a variable to read it easier?

```
isWhitelistEnabled := !parameters.GetBool(parameters.WebAPIAdminWhitelistDisabled)

echoGroup := adm.EchoGroup()

if isWhitelistEnabled {
	echoGroup.Use(protected(adminWhitelist))
}
```

File: tools/devnet/goshimmer.config.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    "webapi": {
+      "basic_auth": {
+        "enabled": false,
+        "username": "goshimmer",
+        "password": "goshimmer"
+      },
+      "bindAddress": "0.0.0.0:8080"
+    },
+    "faucet": {
+      "powDifficulty": 12,
```
Comment (jorgemmsilva) [inline tools/devnet/goshimmer.config.json:83]: shouldn't this be -1 or 0?

**Review by jorgemmsilva — APPROVED** [review]

File: tools/devnet/goshimmer.config.json (hunk 2 above)
Comment (lmoe) [reply tools/devnet/goshimmer.config.json:83]: This is the config for GoShimmer. -1 was set on the waspcli side to use the difficulty of GoShimmer.

**Pull request merged**